docs/cpp/source/notes/tensor_cuda_stream.rst

```diff
@@ -0,0 +1,277 @@
+    tensor1.sum();
+  }
+
+  // current device is reset to device 0 after device_guard is destroyed
+  
+  // acquire a new CUDA stream on device 1
+  at::cuda::CUDAStream myStream1_1 = at::cuda::getStreamFromPool(false, 1);
+  // create a new tensor on device 1
+  torch::Tensor tensor1 = torch::ones({2, 2}, torch.device({torch::kCUDA, 1}));
+
+  // change the current device index to 1 and current CUDA stream on device 1 
+  // to `myStream1_1` using CUDA stream guard within bracket scope
+  {
+    at::cuda::CUDAStreamGuard stream_guard(myStream1_1);
+    // sum() on tensor1 use `myStream1_1` as current CUDA stream on device 1
+    tensor1.sum(); 
+  }
+
+  // current device is reset to device 0 and current CUDA stream on device 1 is 
+  // reset to `myStream1`
+  
+  // sum() on tensor1 uses `myStream1` as current CUDA stream on device 1
+  tensor1.sum();
+
+
+3. Working with CUDA multistream guard
+
+.. code-block:: cpp
+
+  // This example shows how to use CUDA multistream guard to set
+  // two streams on two devices at the same time.
+ 
+  // create two tensor, one on device 0, one on device 1
+  torch::Tensor tensor0 = torch::ones({2, 2}, torch::device({torch::kCUDA, 0}));
+  torch::Tensor tensor1 = torch::ones({2, 2}, torch::device({torch::kCUDA, 1}));
+
+  // acquire new CUDA streams from CUDA stream pool on device 0 and device 1
+  at::cuda::CUDAStream myStream0 = at::cuda::getStreamFromPool(false, 0);
+  at::cuda::CUDAStream myStream1 = at::cuda::getStreamFromPool(false, 1);
+
+  // set current CUDA stream on device 0 to `myStream0` and 
+  // set current CUDA stream on device 1 to `myStream1` CUDA using multistream guard
+  {
+    at::cuda::CUDAMultiStreamGuard multi_guard({myStream0, myStream1});
+
+    // sum() on tensor0 uses `myStream0` as current CUDA stream on device 0
+    tensor0.sum();
+    // sum() on tensor1 uses `myStream1` as current CUDA stream on device 1
+    tensor1.sum();
+  }
+
+  // current CUDA stream on device 0 is reset to default CUDA stream on device 0
+  // current CUDA stream on device 1 is reset to default CUDA stream on device 1 
+
+  // sum() on tensor0 uses default CUDA stream as current CUDA stream on device 0
+  tensor0.sum();
+  // sum() on tensor1 uses defualt CUDA stream as current CUDA stream on device 1
+  tensor1.sum();
+
+.. attention::
+  ``CUDAMultiStreamGuard`` does not change current device index, it only changes the stream on
+  each passed in stream's device. Other than scope controlling, this guard is equivalent to 
+  calling ``setCurrentCUDAStream`` on each passed in stream.
+
+4. A skeleton example for handling CUDA streams on multi-devices
+
+.. code-block:: cpp
+
+   // This is a skeleton example that shows how to handle CUDA streams on multi-devices
+   // Suppose you want to do work on the non-default stream on two devices simultaneously, and we
+   // already have streams on both devices in two vectors. The following code shows possible ways
```

Comment:
"shows possible ways of acquiring, setting CUDA streams." -> "shows three ways of acquiring and setting the streams."